### PR TITLE
WIP: implement RFC 0004-dunder-runner.md

### PR DESCRIPTION
**Pull request by mattp-**
### What does this PR do?
took a cut at adding this via the WithDeprecated system. Thank you to @mirceaulinic for a first cut! this is still a WIP but functionally does what you would want. There is still some work to deal with the runners that are bundled with salt to do the right thing regardless of superseded/not superseded since for this to be merged both paths must work.

### What issues does this PR fix or reference?
implements [rfc/0004](https://github.com/saltstack/salt/blob/develop/rfcs/0004-dunder-runner.md)

### Previous Behavior
`__salt__` in a runner refers to the runner. access to minion mods must be done through instantiating a loader or using salt.cmd runner which in itself is heavyweight and slow.

### New Behavior
when 
```
use_seperseded:
   loader.runner
```
Is added to master config,  `__runner__` becomes the runner dunder, and `__salt__` is the minion dunder. I did not add the optional disable load that was in @mirceaulinic 's original poc, as I think that slow down is from misnamed modules - shouldn't be a problem in develop.

### Tests written?
Not yet, just throwing this up for comments for now

**[NOTICE] Bug fixes or features added to Salt require tests.**
Please review the [test documentation](https://docs.saltstack.com/en/latest/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite.

### Commits signed with GPG?

Yes/No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/loader.py [hunk 1]
```diff
@@ -926,22 +926,59 @@ def call(fun, **kwargs):
...
+    ret.pack['__runner__'] = ret
+    return ret
+
+
+def _runner(opts, utils=None, context=None, whitelist=None):
+    '''
+    Directly call a function inside a loader directory
+    '''
+
+    log.error("there")
```
Comment (gtmanfred) [inline salt/loader.py]: this looks extra

**Review by gtmanfred — APPROVED** [review]

File: salt/loader.py (hunk 1 above)
Comment (mattp-) [reply salt/loader.py]: woops :)

**Review by mirceaulinic — APPROVED** [review]
Looks great, and thanks for taking time to implement this - I dropped the ball here unfortunately. :100:

**Comment by waynew** [comment]
One thing that we would need is some test for this. I still need to familiarize myself with this area of the codebase - not sure who else on @saltstack/team-core would have expertise here to be able to give this implementation a good critique.

**Comment by waynew** [comment]
@mattp- would you be able to write tests for this, and rebase on master?

**Comment by sagetherage** [comment]
@mattp- do you want to rebase this against `Master`?

Discussed in Grooming 2012-APR-19 Considering Closing

**Comment by mattp-** [comment]
@sagetherage i dont think i have time to bring this to test coverage, so I think we can close for now.

**Pull request closed without merging**